### DISCUSSION TITLE
How to replay flows saved in the same session?

### DISCUSSION THREAD

**Discussion by mledezma-r7** [mitmproxy/mitmproxy#7341] [Q&A]
Hey folks, I'm currently saving flows to a file via the `save_stream_file` option, and also using `server_replay` to replay all those flows in the file.

In some cases I receive a request that is not included in the file so I'm appending those flows to it, but then when I get a dupe of that same request the proxy does not replay it because it was not loaded at boot time.

Is there any way to achieve this? I'm using the `save_stream_filter` to just append non replayed flows

**Comment by Prinzhorn** [comment]
Not sure I understand exactly what the problem is, but you can manually work with the saved flows and maybe "patch" them before doing the replay?

https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/io-read-saved-flows.py
https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/io-write-flow-file.py

The replay addon looks like this:

https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/addons/serverplayback.py

and you could also use the commands like `replay.server.add` to add flows to the replay list. Which, looking at the title "in the same session", is likely exactly what you want.
